### ISSUE TITLE
Device did not reconnect. Possibly invalid iBEC. Reset device and try again

### ISSUE THREAD

**Issue by tobia** [LukeZGD/iOS-OTA-Downgrader#168]
Hello

Thank you for putting together this script.

I'm trying to downgrade an iPad Air 1 using a macOS 10.15.7 machine.

I have installed libimobiledevice and libirecovery with homebrew, but it still fails with the error above. It seems to be an error in futurerestore, but I have no idea what to do to change the result.

I have repeated the procedure multiple times, rebooting the device into DFU every time, saved the Terminal logs, and compared them with diff: they are always exactly the same.

Here is the log:

[log.txt](https://github.com/LukeZGD/iOS-OTA-Downgrader/files/8400593/log.txt)

Edit: I also tried downloading the latest futurerestore v2.0.0-test(ecc32003b09a3afb27be80fd881cee9fcc0b5d49-278) as well as using an old v180, replacing resources/tools/futurerestore_macos_x86_64 with them, to no avail.

Any help will be appreciated.

**Comment by LukeZGD** [comment]
Hmm everything seems to check out in the log, it's just that futurerestore doesn't catch the device after booting iBEC for some reason

Try disconnecting and reconnecting the device quickly after the sending iBEC part

**Comment by tobia** [comment]
Thank you for checking my logs.

I'll try that and I'll also try on a different Mac.

**Comment by tobia** [comment]
Nope, still does not reconnect.

I have tried disconnecting and reconnecting the cable quickly after Sending iBEC, during "Booting iBEC, waiting for device to reconnect" to no avail.

I have also tried on a different Mac (Big Sur instead of Catalina) but I get the same error.

What else could it be?

**Comment by LukeZGD** [comment]
Try using a different cable (must be USB A to lightning). If that still doesn't work then I'm not sure on what's happening, but you might want to try booting Linux (after putting the device in pwnDFU mode) and trying the restore from there

**Comment by tobia** [comment]
Nope, I tried 4 different cables, 2 different x86 Macs, and an Ubuntu 20.4 computer (after doing pwnDFU using the Mac)... nothing works!

Are you sure it's not related to these error messages?

```
[Error] [TSSR] Error: could not get id0 for installType=Update
[WARNING] [TSSR] failed to build tssrequest for alternative installType
[TSSR] User specified doesn't to request a baseband ticket.
[TSSR] Request URL set to https://gs.apple.com/TSS/controller?action=2
[TSSR] Sending TSS request attempt 1... failure
[Error] ERROR: TSS request failed (status=128, message=An internal error occurred.)
```

It must be something with my iPad. I'll try resetting / updating it to the latest official iOS and then trying again.

**Comment by tobia** [comment]
Yes, even the official restore procedure fails with the infamous error code "9".

So even though the device goes correctly into restore mode and DFU mode, neither futurerestore nor Apple's own software are able to flash a working image. I brought it to an Apple store and they basically told me to go f*** myself and buy a new one. As if I'd ever buy another product from such a company.

So if anybody wants an ipad4,1 to experiment on, I can donate it for research purposes.

**Comment by tobia** [comment]
So even the official restore procedure (the one that comes up in Finder when you plug in the iPad in recovery or DFU mode) would fail with the infamous error code "9".

I thought the device was soft-bricked, because even though it would restart correctly into recovery mode (with the iTunes logo) and DFU mode, neither futurerestore nor the official restore procedure were able to flash an image. I brought it to an Apple store who told me that it was done for and to buy a new one. Yeah right.

Then, by randomly browsing online, I found a strange hint: **connect a headphone before connecting the USB cable and doing the Finder recovery procedure.** I tried it half-heartedly and **it worked!** Now the device has been restored to its original OS.

I will try again with your script, but I just thought I would mention this strange detail for others who find themselves with this same issue. I wonder if "error 9" means "your device is jailbroken, farewell" and then doing the procedure with headphones plugged in is a backdoor to that lock. One has to wonder...
